### PR TITLE
set expected capacity for file length when init #1135

### PR DESCRIPTION
**Pull request by kaitian521**
ready to be reviewed now.

### CODE REVIEW

**Comment by lingol** [comment]
Hi, there is an ongoing critical issue which requires a hotfix version released once it is fixed. So this PR won't be merged until it's released.

File: Android/MMKV/mmkvdemo/src/main/java/com/tencent/mmkvdemo/MainActivity.java [hunk 1]
```diff
@@ -657,4 +658,28 @@ private void testAutoExpire() {
             Log.e("MMKV", "auto key expiration never_expire_key_1");
         }
     }
+
+    private void testExpectedCapacity() {
+        String key = "key0";
+        String value = "构造函数设置文件大小_";
```
Comment (lingol) [inline Android/MMKV/mmkvdemo/src/main/java/com/tencent/mmkvdemo/MainActivity.java:664]: Better use any UTF-8 alphabet or emoji than zh-cn char.

File: Core/MemoryFile.cpp [hunk 2]
```diff
@@ -212,9 +212,16 @@ void MemoryFile::reloadFromFile() {
         SCOPED_LOCK(&lock);
 
         mmkv::getFileSize(m_diskFile.m_fd, m_size);
+        auto adjustSize = static_cast<size_t>(static_cast<double>(expectedCapacity) * 0.1f);
+        // 8 bytes header + 5 bytes key length + 5 bytes value length + 5 bytes value holder length + 4 bytes expire
+        expectedCapacity += std::max<size_t>(27, adjustSize);
```
Comment (lingol) [inline Core/MemoryFile.cpp:217]: Is this adjustment really necessary? The users should know what they are doing. The `expectedCapacity` they passed in should be a multiply of `DEFAULT_MMAP_SIZE`.

Comment (kaitian521) [reply Core/MemoryFile.cpp]: OK. If the value provided is a multiply of DEFAULT_MMAP_SIZE, it should have no problem here

**Pull request merged**